### DISCUSSION TITLE
PV Forecast - Anlage im nachträglich hinzufügen

### DISCUSSION THREAD

**Discussion by nicodonner** [solectrus/solectrus#2052] [Q&A]
Hallo, 

ich nutze Solectrus schon einige Zeit und bin noch immer begeistert - jedoch habe ich bei der Installation nicht die Daten für den PV Forecast eingetragen. 

Wie kann ich dies nachträglich ändern? 

Ich betreibe Solectrus auf einem PI4 und Docker (Verwaltet über Portainer). 

Vielen Dank 
Gruß
Nico

**Comment by ledermann** [comment]
Die Begeisterung freut mich natürlich :)

Die Angaben zum Forecast (also Geo-Koordinaten, Dachschräge, Ausrichtung etc.) lassen sich jederzeit in der `.env`-Datei ändern. Damit die Änderung wirksam wird, muss nur der Forecast-Collector neu gestartet werden. Das ist am einfachsten über `docker compose down && docker compose up` möglich.

Welche Parameter der Forecast-Collector aktuell verwendet, siehst Du in den Logs des Containers. In der Konsole gib einfach das ein:

```
docker logs solectrus-forecast-collector-1
```

Angezeigt wird die URL mit allen Parametern, die an forecast.solar geschickt wird.

**Comment by Zeppelin-PV** [comment]
eine Frage hierzu
was muss ich tun um in der .env Datei meine Geodaten zu korrigieren ?
Der abgeänderte Wert lässt sich nicht speichern, Datei schreibgeschützt.
Muss ich Solectrus stoppen ?
Wie geht das uf dem PI ?
-LG

**Reply by ledermann** [reply]
Wenn die Datei `.env` schreibgeschützt erscheint, dann stimmt etwas mit den Zugriffsberechtigungen nicht, d.h. der angemeldete User hat keine Schreibberechtigung. Ob SOLECTRUS läuft oder nicht, ist dabei unerheblich.

Auf meinem Raspi 4 sieht es so aus:

```bash
$ whoami
pi

$ ls -l .env
-rw-r--r-- 1 pi pi 3284 Apr 19 07:08 .env
```

Das heißt, ich bin als `pi` angemeldet (das ist der Standard-User auf einem Raspberry Pi), der User `pi` (als Owner dieser Datei) hat Schreibberechtigung auf `.env`, alle anderen nur Leseberechtigung.

Bei der Installation von SOLECTRUS ist die Datei von Dir bearbeitet worden, zumindest musste ja die IP-Adresse Deines Speichers eingetragen werden. Möglicherweise bist Du jetzt mit einem anderen User eingeloggt.

Du kannst mittels `sudo` temporär Superuser-Rechte erhalten. Um eine Datei zu bearbeiten, für die der angemeldete User keine Schreibberechtigung hast, kannst Du folgendes machen:

```bash
sudo nano .env
```

Ansonsten: Dateirechte können geändert werden, ich würde aber aber empfehlen, zunächst die Ursache für die fehlende Schreibberechtigung zu klären. 

Falls Du dennoch die Rechte ändern willst, dann geht das wie folgt, z.B. auf den User `pi` der Usergroup `pi`:

```bash
sudo chown pi:pi .env
```

**Comment by Zeppelin-PV** [comment]
nach ändern mit sudo nano .env läuft solectrus nicht mehr
![grafik](https://github.com/solectrus/solectrus/assets/126392805/fcc27c57-82f1-4c62-a0b5-873fb360f02a)
die .env ist noch da aber mit dem alten Geo Wert
docker compose down und up hilft nicht, 
was habe ich da zerschossen ?

**Reply by ledermann** [reply]
Dem Screenshot lässt sich nicht entnehmen, was da falsch läuft. Interessanter wäre die Ausgabe der Container-Logs bzw. dessen, was bei "docker compose" erscheint.

Auch weiß ich nicht, als welcher User Du eingeloggt bist und welche Berechtigungen vorher für die einzelnen Dateien vergeben waren. Das wäre aber nützlich zu wissen, um den vorherigen Zustand wiederherzustellen. Grundsätzlich kann mit `chown` der Owner und mit `chmod` die Zugriffsberechtigung für eine Datei verändert werden. 

Docker selbst greift als `root` auf alles zu, insofern verstehe ich nicht, warum das jetzt nicht funktionieren soll. Ich brauche da deutlich mehr Input, um helfen zu können.

**Reply by ledermann** [reply]
Zu welchem Output führen folgende Konsolen-Befehle bei Dir?

```bash
cd /home/pi/solectrus
whoami
ls -la
```

**Comment by Zeppelin-PV** [comment]
admin@raspberrypi:/home/pi/solectrus $ whoami
admin
admin@raspberrypi:/home/pi/solectrus $ ls -la
insgesamt 32
drwxr-xr-x  6 root  root 4096 26. Feb 19:21 .
drwxr-xr-x  3 root  root 4096 25. Feb 17:15 ..
drwxr-xr-x  2 root  root 4096 25. Feb 22:29 csv
-rw-r--r--  1 root  root 2531 25. Feb 21:31 docker-compose.yml
-rw-r--r--  1 root  root 3201 25. Feb 21:30 .env
drwx------  3 admin root 4096 25. Feb 17:57 influxdb
drwx------ 19    70 root 4096  3. Jun 08:20 postgresql
drwxr-xr-x  2   999 root 4096  3. Jun 08:20 redis
admin@raspberrypi:/home/pi/solectrus $

**Comment by ledermann** [comment]
Du bist also mit dem User `admin` angemeldet, nicht als `pi`. Insofern ist die Nutzung des Ordners `/home/pi` etwas unglücklich, aber das sollte nichts ausmachen.

Zu sehen ist außerdem, dass die `.env` seit Februar nicht mehr verändert wurde. Aber das ist jetzt wohl eher sekundär, es geht darum, SOLECTRUS wieder ans laufen zu bekommen.

Was passiert denn beim Versuch, SOLECTRUS zu starten? Was ist der Output von `docker compose up -d`?
Ansonsten: Laufen die Container danach? Was ergibt `docker ps`?

Ohne genaue Beschreibung dessen, was Du gemacht hast und was das Ergebnis ist, kommen wir nur schwerlich weiter.

**Comment by Zeppelin-PV** [comment]
admin@raspberrypi:/home/pi/solectrus $ docker compose up -d
permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/json?all=1&filters=%7B%22label%22%3A%7B%22com.docker.compose.config-hash%22%3Atrue%2C%22com.docker.compose.project%3Dsolectrus%22%3Atrue%7D%7D": dial unix /var/run/docker.sock: connect: permission denied
admin@raspberrypi:/home/pi/solectrus $ 


admin@raspberrypi:/home/pi/solectrus $ docker ps
permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/json": dial unix /var/run/docker.sock: connect: permission denied
admin@raspberrypi:/home/pi/solectrus $

**Comment by ledermann** [comment]
Du hast vermutlich mehr verändert, als Du geschrieben hast. Dem User `admin` ist offensichtlich der Zugriff auf Docker untersagt. Es dürfte helfen, wenn Du jedem docker-Befehl ein `sudo` voranstellst, also z.B. so:

```bash
sudo docker compose up -d 
```

Dazu aus der [SOLECTRUS-Installationsanleitung](https://github.com/solectrus/hosting/tree/main/guide/raspberry-pi):

> Don't forget to add your user to the docker group, so you don't need to use sudo for every Docker command.
>
>```bash
>sudo groupadd docker
>sudo usermod -aG docker $USER
>```
>
> Log out and log back in so that your group membership is re-evaluated.
>
> Details: https://docs.docker.com/engine/install/linux-postinstall/

**Comment by Zeppelin-PV** [comment]
Entwarnung !!!
Solectrus läuft. Vermutlich sogar gleich nach dem ändern des Azimuthwertes.
Firefox hat mir ne Falle gestellt mit https:\

ich hatte in dem Azimuthwert -93 eingegeben
das Haus steht aber 3 Grad links von Norden ( in einer 360er Anzeige also bei 357)
dann müssten es 177 sein, oder. Die Forecast Einteilung ist etwas gewöhnungsbedürftig.
Eine große Veränderung sehe ich trotzdem nicht.
... egal , es läuft wieder
DANKE Georg

**Reply by ledermann** [reply]
Ok, klingt gut, auch wenn ich dann die Fehlermeldungen aus der SSH-Konsole nicht verstehe.

Bezüglich Azimuth gibt es beim Betreiber `forecast.solar` noch eine nützliche Info-Seite:
https://doc.forecast.solar/find_your_azimuth

**Comment by Zeppelin-PV** [comment]
Gedankenfehler, -93 war doch richtig
die Hausachse Nord-Süd ist 357 Nord, 177 Süd (360Grad Teilung)
somit ist die Ostfläche 87 bei der 360er Teilung, und -93 bei der Forcast Teilung
